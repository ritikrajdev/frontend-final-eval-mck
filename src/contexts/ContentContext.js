import PropTypes from 'prop-types';
import React, { createContext } from 'react';

export const ContentContext = createContext([]);

export default function ContentContextProvider({ children }) {
  const [content, setContent] = React.useState(null);

  return (
    <ContentContext.Provider value={[content, setContent]}>
      {children}
    </ContentContext.Provider>
  );
}

ContentContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
