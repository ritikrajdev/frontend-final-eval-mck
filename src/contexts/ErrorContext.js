import PropTypes from 'prop-types';
import React, { createContext } from 'react';

export const ErrorContext = createContext([]);

export default function ErrorContextProvider({ children }) {
  const [error, setError] = React.useState(null);

  if (error) {
    setTimeout(() => {
      setError(null);
    }, 5000);
  }

  return (
    <ErrorContext.Provider value={[error, setError]}>
      {error && (
        <div
          style={{
            position: 'absolute',
            bottom: '20px',
            left: '20px',
            backgroundColor: 'white',
            display: 'inline-block',
            padding: '20px',
            borderRadius: '8px',
            color: 'red',
            border: '1px solid black',
          }}
        >
          {error}
        </div>
      )}
      {children}
    </ErrorContext.Provider>
  );
}

ErrorContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
