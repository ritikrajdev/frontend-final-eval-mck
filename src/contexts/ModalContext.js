import PropTypes from 'prop-types';
import React, { createContext } from 'react';
import Button from '../components/elements/Button';
import Input from '../components/elements/Input';

export const ModalContext = createContext([]);

export default function ModalContextProvider({ children }) {
  const [modal, setModal] = React.useState({
    // position: 'center' | 'right'
    show: false,
    position: 'center',
    title: 'Modal',
    fields: {},
    onSave: () => {
      return true;
    },
  });

  let positionStylingProps = {};

  if (modal.position === 'center') {
    positionStylingProps = {
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    };
  } else if (modal.position === 'right') {
    positionStylingProps = {
      top: '0',
      right: '0',
      bottom: '0',
    };
  }

  return (
    <ModalContext.Provider value={[modal, setModal]}>
      {modal.show && (
        <div
          style={{
            width: '100%',
            height: '100vh',
            backgroundColor: '#00000099',
            position: 'absolute',
          }}
          onClick={() => {
            setModal({ ...modal, show: false });
          }}
        >
          <form
            style={{
              ...positionStylingProps,
              position: 'absolute',
              backgroundColor: 'white',
              display: 'inline-block',
              padding: '20px',
              borderRadius: '8px',
              color: 'red',
              border: '1px solid black',
            }}
            onClick={(e) => e.stopPropagation()}
            onSubmit={(e) => {
              e.preventDefault();
              if (modal.onSave(modal.fields))
                setModal({ ...modal, show: false });
            }}
          >
            <h2>{modal.title}</h2>
            <br />
            {Object.entries(modal.fields).map(([key, value]) => (
              <div key={key}>
                <Input
                  label={key}
                  value={value}
                  onChange={(e) =>
                    setModal({
                      ...modal,
                      fields: { ...modal.fields, [key]: e.target.value },
                    })
                  }
                />
                <br />
              </div>
            ))}

            <center>
              <Button type='submit'>Save</Button>
            </center>
          </form>
        </div>
      )}
      {children}
    </ModalContext.Provider>
  );
}

ModalContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
