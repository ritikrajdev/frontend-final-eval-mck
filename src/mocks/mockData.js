export const mockCollections = [
  {
    id: 5,
    name: 'Laptops',
    schema: {
      brand: 'text',
    },
    userId: 1,
    formResponses: [
      {
        id: 8,
        form_id: 5,
        response: {
          brand: 'dell',
        },
      },
      {
        id: 4,
        form_id: 5,
        response: {
          brand: 'apple',
        },
      },
    ],
  },
];
