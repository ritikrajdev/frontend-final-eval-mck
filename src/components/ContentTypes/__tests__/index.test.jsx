import { render } from '@testing-library/react';
import React from 'react';
import { useNavigate } from 'react-router-dom';
import ContentType from '..';
import { ContentContext } from '../../../contexts/ContentContext';
import { ErrorContext } from '../../../contexts/ErrorContext';
import { mockCollections } from '../../../mocks/mockData';

jest.mock('react-router-dom');

describe('ContentType', () => {
  it('should render correctly', async () => {
    const navigate = jest.fn();
    useNavigate.mockReturnValue(navigate);

    const error = 'some error';
    const setError = jest.fn();

    const collections = mockCollections;
    const setCollections = jest.fn();

    const { asFragment } = render(
      <ErrorContext.Provider value={[error, setError]}>
        <ContentContext.Provider value={[collections, setCollections]}>
          <ContentType collectionId={mockCollections[0].id} />
        </ContentContext.Provider>
      </ErrorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
