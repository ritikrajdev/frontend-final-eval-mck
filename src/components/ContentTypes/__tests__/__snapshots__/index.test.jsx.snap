// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    class="content-types"
  >
    <h2>
      <span>
        Content Types
      </span>
       
      <button
        class="sign-out-button"
      >
        Sign Out
      </button>
    </h2>
    <div
      class="types-container"
    >
      <div
        class="left-container"
      >
        <p>
          1 Types
        </p>
        <br />
        <button
          class="add-button"
        >
          + New Type
        </button>
        <div>
          <button
            class="content-type-button"
          >
            <span>
              Laptops
            </span>
            <span>
              1
            </span>
          </button>
        </div>
      </div>
      <div
        class="right-container"
      />
    </div>
  </div>
</DocumentFragment>
`;
